app/controllers/students_controller.rb

```diff
@@ -30,6 +29,11 @@ def show
     }
   end
 
+  def restricted_notes
+    raise Exceptions::EducatorNotAuthorized unless current_educator.can_view_restricted_notes
+    show(restricted_notes: true)
```

Comment:
It's up to you but my initial read is that the code in the previous `restricted_notes` method is more direct.  Calling the same queries as the `show` action pulls in a bunch of data that isn't used on the restricted notes page.  This is fine from a performance perspective, but is probably unexpected in terms of "why is there chart data on the restricted notes page?"  And with your great change to add the `restricted_notes` param to `student_feed`, we can use that get what we need in the `restricted_notes` action and that's only a few lines long.